pkg/roachprod/vm/aws/aws.go

```diff
@@ -706,60 +709,81 @@ func (p *Provider) Create(
 		res := limiter.Reserve()
 		g.Go(func() error {
 			time.Sleep(res.Delay())
-			return p.runInstance(l, capName, index, placement, machineType, opts, providerOpts)
+			vm, err := p.runInstance(l, capName, index, placement, machineType, opts, providerOpts)
+			if err != nil {
+				return err
+			}
+
+			vmListLock.Lock()
+			defer vmListLock.Unlock()
+			vmList = append(vmList, *vm)
+
+			return nil
 		})
 	}
 	if err := g.Wait(); err != nil {
-		return err
+		return nil, err
+	}
+
+	// Our initial list of VMs does not include the IP addresses or volumes.
+	// waitForIPs() returns the VMs list with all information set, so we simply
+	// overwrite the list with the result of waitForIPs().
```

Comment:
Why keep track of `vmList` above if we will always overwrite it here?

Reply:
A bit of refactoring gone wrong, thanks!